.vscode/settings.json

```diff
@@ -27,5 +27,9 @@
   "[typescript]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode",
     "editor.formatOnSave": true
+  },
+  "[typespec]": {
+    "editor.defaultFormatter": "typespec.typespec-vscode",
```

Comment:
Is this formatter pre-installed by the `package-lock.json` in repo root, or is this something devs will need to install manually?

Reply:
this is the typespec vscode extension that provide the format action.
The extension then use the language server provided in the locally installed compiler

Reply:
Will it work like this?

1. Devs can optionally choose to install the typespec vscode extension
2. If installed, this enables formatOnSave
3. In not installed, this is a no-op

Reply:
on save it is a noop if the extension is not here, if you explicitly do format then it shows this popup
![image](https://github.com/Azure/azure-rest-api-specs/assets/1031227/a49a064a-c7fc-4b21-bca5-190a932b9bc5)

Reply:
should actually add the typespec extension as a recommendation i'll do that